### DISCUSSION TITLE
Add a 'previousRoute' property to the Router instance

### DISCUSSION THREAD

**Discussion by nathanchase** [vuejs/router#1791] [Ideas]
### What problem is this solving

I'd like to see a `previousRoute` property available on the Router instance.

This would be the value of where the router was **_previously_** - (`undefined`) if there is no such value.

This is an upstream request that came from needing this property on the Nuxt 3 router:
https://github.com/nuxt/nuxt/issues/15661

### Proposed solution

Add a 'previousRoute' property to the Router instance.

### Describe alternatives you've considered

Currently, I have to create a global middleware to store the previous route in Pinia, like so:

```
export default defineNuxtRouteMiddleware(async (to, from) => {
  const globalsStore = useGlobalsStore();
  if (from.name) {
    globalsStore.previousRoute = from.name.toString();
  }
});
```

**Comment by posva** [comment]
This could be useful although it can already be added with a router guard:

```ts
router.beforeEach((to, from) => {
  to.meta.from = to
})
```

So the question would be if there are enough common use cases for this to be a default feature in router. FYI, this has been requested before, so it's worth checking issues in vuejs/router and vuejs/vue-router
